include/nbl/ext/LumaMeter/CLumaMeter.h

```diff
@@ -118,13 +130,67 @@ class CLumaMeter : public core::TotalInterface
+			video::IGPUDescriptorSet* ds,
+			const core::smart_refctd_ptr<video::IGPUBuffer> paramUbo,
+			const asset::SBufferRange<video::IGPUBuffer>& outputSSBORange,
+			const core::smart_refctd_ptr<video::IGPUImageView> inputImageView,
+			video::ILogicalDevice* logicalDevice)
 		{
-			driver->dispatch(dispatchInfo.workGroupCount[0], dispatchInfo.workGroupCount[1], dispatchInfo.workGroupCount[2]);
+			static_assert(mode < EMM_COUNT, "Invalide meter mode!");
+
+			video::IGPUDescriptorSet::SWriteDescriptorSet writes[DEFAULT_DESCRIPTOR_COUNT] = {};
+			video::IGPUDescriptorSet::SDescriptorInfo infos[DEFAULT_DESCRIPTOR_COUNT] = {};
+
+			const asset::E_DESCRIPTOR_TYPE descriptorTypes[DEFAULT_DESCRIPTOR_COUNT] =
+			{
+				asset::EDT_UNIFORM_BUFFER, // luma input params
+				asset::EDT_STORAGE_BUFFER, // luma output buffer
+				asset::EDT_COMBINED_IMAGE_SAMPLER // input image
+			};
+
+			infos[0].desc = paramUbo;
+			infos[0].buffer.offset = 0ull;
+			infos[0].buffer.size = sizeof(Uniforms_t<mode>);
+
+			infos[1].desc = outputSSBORange.buffer;
+			infos[1].buffer.offset = outputSSBORange.offset;
+			infos[1].buffer.size = outputSSBORange.size - infos[1].buffer.offset;
 
-			if (issueDefaultBarrier)
-				defaultBarrier();
+			infos[2].desc = inputImageView;
+			infos[2].image.sampler = nullptr;
+			infos[2].image.imageLayout = asset::EIL_SHADER_READ_ONLY_OPTIMAL;
+
+			for (uint32_t binding = 0u; binding < DEFAULT_DESCRIPTOR_COUNT; ++binding)
+			{
+				writes[binding].dstSet = ds;
+				writes[binding].binding = binding;
+				writes[binding].arrayElement = 0u;
+				writes[binding].count = 1u;
+				writes[binding].descriptorType = descriptorTypes[binding];
+				writes[binding].info = infos + binding;
+			}
+
+			logicalDevice->updateDescriptorSets(DEFAULT_DESCRIPTOR_COUNT, writes, 0u, nullptr);
+		}
+
+		// we expect user binds correct pipeline, descriptor sets and pushes the push constants by themselves
+		static inline void dispatchHelper(
+			video::IGPUCommandBuffer* cmdbuf, 
+			const DispatchInfo_t& dispatchInfo,
+			const asset::E_PIPELINE_STAGE_FLAGS srcStageMask,
+			const uint32_t srcBarrierCount,
+			const video::IGPUCommandBuffer::SBufferMemoryBarrier* srcBarriers,
+			const asset::E_PIPELINE_STAGE_FLAGS dstStageMask,
+			const uint32_t dstBarrierCount,
+			const video::IGPUCommandBuffer::SBufferMemoryBarrier* dstBarriers)
+		{
+			if (srcStageMask != asset::E_PIPELINE_STAGE_FLAGS::EPSF_TOP_OF_PIPE_BIT && srcBarrierCount)
+				cmdbuf->pipelineBarrier(srcStageMask, asset::EPSF_COMPUTE_SHADER_BIT, asset::EDF_NONE, 0u, nullptr, srcBarrierCount, srcBarriers, 0u, nullptr);
+			cmdbuf->dispatch(dispatchInfo.workGroupCount[0], dispatchInfo.workGroupCount[1], dispatchInfo.workGroupCount[2]);
+			if (dstStageMask != asset::E_PIPELINE_STAGE_FLAGS::EPSF_BOTTOM_OF_PIPE_BIT && dstBarrierCount)
+				cmdbuf->pipelineBarrier(asset::EPSF_COMPUTE_SHADER_BIT, dstStageMask, asset::EDF_NONE, 0u, nullptr, dstBarrierCount, dstBarriers, 0u, nullptr);
```

Comment:
could we have a useful overload without the barrier arguments or provide some sane defaults?

Reply:
@nipunG314 the barrier API changed quite a bit, we don't really know the buffers and images the user will use

the source and dest pipeline stage flags are now inside the buffer and image memory barriers

so `dispatchHelper` has no reason to exit anymore